scripts/validate-chart-versions.sh

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+GIT_REPO_ROOT=$(git rev-parse --show-toplevel)
+BUILD_DIR="${GIT_REPO_ROOT}/chartbuild"
+TOOLS_DIR="${BUILD_DIR}/tools"
+CHARTS="${GIT_REPO_ROOT}/deploy/charts"
+export PATH="${TOOLS_DIR}:${PATH}"
+
+EXIT_CODE=0
+if $(git describe HEAD --tags | grep -Eq "^v[0-9]+(\.[0-9]+)*(-[a-z0-9]+)?$"); then
+    LAST_RELEASE_HASH=$(git rev-list --tags --max-count=1 --skip=1 --no-walk)
+else 
+    TAG=$(git describe HEAD --tags | grep -Eo "^v[0-9]+(\.[0-9]+)*")
+    LAST_RELEASE_HASH=$(git rev-list -1 $TAG)
+fi
+LAST_RELEASE_TAG=$(git describe $LAST_RELEASE_HASH --tags)
+cd ${CHARTS}
+echo "📝 Checking for updated Chart versions since the last brupop release $LAST_RELEASE_TAG"
+for d in */; do
+    LAST_COMMIT_HASH=$(git --no-pager log --pretty=tformat:"%H" -- $d | awk 'FNR <= 1')
+    ## If LAST_RELEASE_HASH does not include the chart, then it's a new chart and does not need a version increment
+    if [[ -z $(git ls-tree -d $LAST_RELEASE_HASH $d) ]]; then
+    echo "✅ Chart $d is a new chart since the last release"
+    continue
+    fi
+    ## If LAST_RELEASE_HASH is NOT an ancestor of LAST_COMMIT_HASH then it has not been modified 
+    if [[ ! -z $LAST_COMMIT_HASH && -z $(git rev-list $LAST_COMMIT_HASH | grep $LAST_RELEASE_HASH) || $LAST_COMMIT_HASH == $LAST_RELEASE_HASH ]]; then
```

Comment:
That's right, it would require an uncommitted chart.

I think this is probably more relevant for EKS who maintain many charts, but I don't necessarily see the harm in leaving it here.

Reply:
I reworked the function to make this case more clear and exit early.

Reply:
Oh, yes, you mentioned it being based on the EKS scripts, but I actually reviewed from scratch and didn't look at the originals. No harm in leaving it in at all, just been wondering to check my understanding.